### DISCUSSION TITLE
Tree : Expand children on label click

### DISCUSSION THREAD

**Discussion by AlexyNau** [primefaces/community#368] [PrimeVue] [answered]
Hello,
It doesn't seems it has been requested so I try :

## Idea
In Tree component (Data > Tree), adding the possibility to expand a node's children on label click.

## Conditions
It would append with in these conditions :

- Current node at least one child (`this.node.children && this.node.children.length > 0`)
- Selection mode is defined (`selectionMode != null`)
- Node is not selectable (`this.node.selectable === false`)

## Example
Here is an example where only the last child is selectable :

| ![image](https://github.com/primefaces/community/assets/36192838/df93cb7b-3ec4-4c77-b962-d2b0674132bd)  | Click on 'Documents' : ![image](https://github.com/primefaces/community/assets/36192838/79e3117a-1830-4973-87da-6d26c2004eae) |
| ------------- | ------------- |
| Click on 'Works' : ![image](https://github.com/primefaces/community/assets/36192838/982755c9-4c49-4303-b921-29afbcc9ef7e)  | Select a child : ![image](https://github.com/primefaces/community/assets/36192838/5ac890ae-46b3-472b-9eab-338ff41eda72)  |

So here, in every parent Node, I have set `selectable: false,`

_Note :_ This feature will work fine with `selectionMode` to `single` or `multiple`. But with `checkbox`, today, a click on label select all selectable children. So it can be decided to enabled this feature only for `single` and `multiple`, or also for `checkbox` with the drawback having an undesirable selection/unselection when opening nodes. 
Example when I open **Documents** then **Work**  (Only Documents and Works nodes have `selectable: false`)

![image](https://github.com/primefaces/community/assets/36192838/94ce89fe-2e24-4965-9d83-0ce1a74fc8eb)

Do you think it could be an enhancement to Tree component ? Or maybe find a way to have `checkbox` mode working fine with this feature ?

Thanks for your feedbacks,

**Comment by tugcekucukoglu** [comment, marked as answer]
Not only the Tree component, but both TreeSelect and TreeTable components share the same idea. Whether in selection mode or not, subnodes become visible when the toggler button is clicked, not when the node itself is clicked. This means a huge change in terms of backward compatibility and community feedback is needed for this.

Thank you for your interest!

**Reply by ogunsakin01** [reply]
Finally have this same problem, will be nice if primeVue support this out of the box or reference a doc on how to implement
